ksqldb-parser/src/main/java/io/confluent/ksql/util/QueryAnonymizer.java

```diff
@@ -0,0 +1,776 @@
+import io.confluent.ksql.parser.SqlBaseParser.ListVariablesContext;
+import io.confluent.ksql.parser.SqlBaseParser.LogicalBinaryContext;
+import io.confluent.ksql.parser.SqlBaseParser.PartitionByContext;
+import io.confluent.ksql.parser.SqlBaseParser.PrintTopicContext;
+import io.confluent.ksql.parser.SqlBaseParser.QueryContext;
+import io.confluent.ksql.parser.SqlBaseParser.RegisterTypeContext;
+import io.confluent.ksql.parser.SqlBaseParser.SelectItemContext;
+import io.confluent.ksql.parser.SqlBaseParser.SelectSingleContext;
+import io.confluent.ksql.parser.SqlBaseParser.SetPropertyContext;
+import io.confluent.ksql.parser.SqlBaseParser.ShowColumnsContext;
+import io.confluent.ksql.parser.SqlBaseParser.SingleExpressionContext;
+import io.confluent.ksql.parser.SqlBaseParser.SingleStatementContext;
+import io.confluent.ksql.parser.SqlBaseParser.StatementsContext;
+import io.confluent.ksql.parser.SqlBaseParser.StringLiteralContext;
+import io.confluent.ksql.parser.SqlBaseParser.TableElementContext;
+import io.confluent.ksql.parser.SqlBaseParser.TableElementsContext;
+import io.confluent.ksql.parser.SqlBaseParser.TablePropertiesContext;
+import io.confluent.ksql.parser.SqlBaseParser.TablePropertyContext;
+import io.confluent.ksql.parser.SqlBaseParser.TerminateQueryContext;
+import io.confluent.ksql.parser.SqlBaseParser.TypeContext;
+import io.confluent.ksql.parser.SqlBaseParser.UndefineVariableContext;
+import io.confluent.ksql.parser.SqlBaseParser.UnquotedIdentifierContext;
+import io.confluent.ksql.parser.SqlBaseParser.UnsetPropertyContext;
+import io.confluent.ksql.parser.SqlBaseParser.ValueExpressionContext;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.apache.commons.lang3.StringUtils;
+
+public class QueryAnonymizer {
+
+  public String anonymize(final String query) {
+    final ParserRuleContext tree = getParseTree(query);
+    return build(tree);
+  }
+
+  private ParserRuleContext getParseTree(final String query) {
+    return DefaultKsqlParser.getParseTree(query);
+  }
+
+  private String build(final ParserRuleContext parseTree) {
+    return new Visitor().visit(parseTree);
+  }
+
+  private static final class Visitor extends SqlBaseBaseVisitor<String> {
+    private int streamCount = 1;
+    private int columnCount = 1;
+    private int tableCount = 1;
+    private final Hashtable<String, String> anonTable = new Hashtable<>();
+
+    @Override
+    public String visitStatements(final StatementsContext context) {
+      final List<String> statementList = new ArrayList<>();
+      for (final SingleStatementContext stmtContext : context.singleStatement()) {
+        final String statement = visitSingleStatement(stmtContext);
+        statementList.add(statement);
+      }
+      return statementList.get(0);
```

Comment:
When writing this I assumed that it will only ever be one statement but you are right, there might be instances where we anonymize multiple statements at once. I updated it to return a semicolon separated string will all anonymized statements instead :)